### PR TITLE
Sort plugins before codegen

### LINKED ISSUE #92: generated code changes when plugins are compiled with go1.5
Looks like this is because plugins are registered using `init()`, and the load order has changed in 1.5, but I'm not sure.

### PR DESCRIPTION
**Pull request by tamird**
Fixes #92.

This diff is **big**. Sorry about that.

### CODE REVIEW

**Comment by BramGruneir** [comment]
LGTM

**Comment by awalterschulze** [comment]
Could you explain why the sort is needed.
I am a bit tired, but I don't see any reason that the plugin list would not be deterministic.
Yes I still have not installed go1.5, but just looking at the code, I can't see what is going wrong.

**Comment by awalterschulze** [comment]
Basically I think the change is fine, but I just want to understand exactly why it is needed first.

**Comment by tamird** [comment]
The order of init() calls seems to have changed in 1.5.

**Comment by tamird** [comment]
To clarify, it is still deterministic in 1.5, but the order differs from
1.4's. This ensures that both versions will produce the same code.

File: protoc-gen-gogo/generator/generator.go [hunk 1]
```diff
@@ -77,7 +78,21 @@ type Plugin interface {
...
+
+func (ps pluginSlice) Less(i, j int) bool {
+	return ps[i].Name() < ps[j].Name()
+}
+
+func (ps pluginSlice) Swap(i, j int) {
+	ps[i], ps[j] = ps[j], ps[i]
+}
+
+var plugins pluginSlice
```
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: This can change back to []Plugin

**Comment by awalterschulze** [comment]
Everything else looks good and I understand now.   For some reason I thought the range was happening over pluginList and not plugins, as I said I am tired.

**Comment by awalterschulze** [comment]
Thanks again for this fix :)

**Comment by tamird** [comment]
My pleasure!

**Pull request merged**